src/phongo_bson_encode.c

```diff
@@ -58,6 +58,10 @@ static int php_phongo_is_array_or_document(zval* val)
 	int        count;
 
 	if (Z_TYPE_P(val) != IS_ARRAY) {
+		if (Z_TYPE_P(val) == IS_OBJECT && instanceof_function(Z_OBJCE_P(val), php_phongo_packedarray_ce)) {
+			return IS_ARRAY;
+		}
+
```

Comment:
While we're here, I think there's some redundant code later in this function if you'd like to address it now.

```c
if (count > 0) {
    ...
} else {
    return Z_TYPE_P(val);
}

return IS_ARRAY;
```

After this conditional for `(Z_TYPE_P(val) != IS_ARRAY)`, we can rely on the final `return IS_ARRAY` unless iteration of a non-empty array indicates that we need to return `IS_OBJECT`. The `else` branch above is redundant.

This can also wait until we refactor this function in [PHPC-2217](https://jira.mongodb.org/browse/PHPC-2217), so I've left a comment there as well.

Reply:
Left this to be handled in PHPC-2217.